File: simon-nystrom-python3/day2.py
file = open("input-2.txt")
chart = [x.strip() for x in file.readlines()]
file.close()


def solve1(chart):
    result = 0
    for l in chart:
        row = [int(x) for x in l.split('\t')]
        result += max(row) - min(row)
    return result


def solve2(chart):
    result = 0
    for l in chart:
        row = [int(x) for x in l.split('\t')]
        result += row_result(row)

    return result


def row_result(row):
    for i, x in enumerate(row):
        for j, y in enumerate(row):
            if x % y == 0 and i != j:
                return x // y
            elif y % x == 0 and i != j:
                return  y // x


print(solve1(chart))
print(solve2(chart))

File: ryggan-JavaScript/day2.js
import fs from 'fs';
const input = fs.readFileSync('./day2input.txt', 'utf8');

const rows = input.split('\n');

const sum1 = rows.reduce((acc, row) => {
  const rowResult = row.split('\t').reduce((acc, n) => {
    n = Number(n);
    if (n < acc.min) {
      acc.min = n;
    }
    if (n > acc.max) {
      acc.max = n;
    }
    return acc;
  }, { min: 10000, max: 0 });
  const { min, max } = rowResult;
  return acc + (max - min);
}, 0);

console.log('Uppgift 1: ' + sum1);

let t = 0;
const sum2 = rows.reduce((acc, row) => {
  const rowResult = row.split('\t').reduce((acc, n) => {
    n = Number(n);
    let div = -1;
    row.split('\t').map(d => {
      d = Number(d);
      if (d < n && n % d === 0) {
        div = d;
      }
    });

    if (div === -1) {
      return acc;
    }
    return (acc = n / div);
  }, 0);
  return (acc += rowResult);
}, 0);

console.log('Uppgift 2: ' + sum2);
